src/core/border_router/routing_manager.cpp

```diff
@@ -641,26 +641,47 @@ void RoutingManager::EvaluateNat64Prefix(void)
 
     NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
     NetworkData::ExternalRouteConfig config;
-    bool                             hasAdvertisedNat64Prefix = false;
+    const Ip6::Prefix *              smallestNat64Prefix = nullptr;
 
     otLogInfoBr("Evaluating NAT64 prefix");
 
     while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
     {
-        if (config.mNat64 && config.GetPrefix().IsValidNat64())
+        const Ip6::Prefix &prefix = config.GetPrefix();
+
+        if (config.mNat64 && prefix.IsValidNat64())
         {
-            hasAdvertisedNat64Prefix = true;
+            if (smallestNat64Prefix == nullptr || (prefix < *smallestNat64Prefix))
+            {
+                smallestNat64Prefix = &prefix;
+            }
         }
     }
 
-    if (!hasAdvertisedNat64Prefix && !mIsAdvertisingLocalNat64Prefix)
+    if (smallestNat64Prefix == nullptr)
```

Comment:
I think we also need to `AddExternalRoute` if `smallestNat64Prefix` exists and (*smallestNat64Prefix = mLocalNat64Prefix). 

It's possible that the local NAT64 prefix is in the Leader's network data, but not locally advertised yet (e.g. when the BR is rebooted and just restored the Network Data). In such case, the BR should re-add the NAT64 prefix to the local Network Data.

Thoughts? @xuyirio @wgtdkp @librasungirl

Reply:
SGTM.

Reply:
Updated. PTAL. Thanks!